doc/TROUBLESHOOTING.md

```diff
@@ -145,3 +145,69 @@ Mountpoint logs should contain an error similar to below:
 WARN setattr{req=4 ino=21 name="init.txt"}:
 mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "init.txt") is a remote inode and its attributes cannot be modified
 ```
+
+## Invalid Hostname for DNS resolution
+
+Mountpoint by default resolves endpoint for requests in [virtual hosted style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html).
+If your storage provider does not support virtual style hosted bucket, you may recieve the following error:
+
+```
+Error: Failed to create S3 client
+
+Caused by:
+    0: initial ListObjectsV2 failed for bucket my-bucket in region us-east-1
+    1: Client error
+    2: Unknown CRT error
+    3: CRT error 1059: aws-c-io: AWS_IO_DNS_INVALID_NAME, Host name was invalid for dns resolution.
+Error: Failed to create mount process
+```
+
+In this case, try using `--force-path-style` CLI option when you are mounting the bucket using Mountpoint.
```

Comment:
Basically, _virtual hosted style_ is `https://docexamplebucket.example.com`, compared to _path style_ being `https://example.com/docexamplebucket`. I was only able to comprehend it after reading through our [CONFIGURATION.md](https://github.com/awslabs/mountpoint-s3/blob/82a5fe606c047fee7e7015537cde0b4aabcf0133/doc/CONFIGURATION.md#endpoints-and-aws-privatelink). 

Do you think adding an example (or a link to `CONFIGURATION.md` with such example) will help?

Reply:
I have already added the link to aws documentation of virtual hosted style addressing. In that difference is explained well with example.